config/config.go

```diff
@@ -166,8 +165,11 @@ func (cfg Global) Validate() error { //nolint:gocognit
 		appendError(fmt.Errorf("-requests: must be >= 0, we got %d", cfg.Runner.Requests))
 	}
 
-	if cfg.Runner.Concurrency < 1 && cfg.Runner.Concurrency != -1 {
-		appendError(fmt.Errorf("-concurrency: must be > 0, we got %d", cfg.Runner.Concurrency))
+	if cfg.Runner.Concurrency < 1 || cfg.Runner.Concurrency > cfg.Runner.Requests {
```

Comment:
Technically having more available threads than requests to run is not a problem.

Example:
- at most 10 concurrent threads
- do 5 requests
- at 5, the for loop exits, the dispatchers doesn't care, not its job to know it will not be used fully

Reply:
This change is an alignment with how `dispatcher` _already_ works, so we display an insightful error message.
If we want to change that let's do it in a dedicated discussion + issue + PR!